### PR TITLE
Feature/feature sets

### PR DESCRIPTION
**Pull request by ArthurDeclercq**
- Replaced current feature sets with a list of lists with each list containing the features included in one pin file for rescoring
- Rewrote plotting.py with classes 
- added the option of plotting some rescore plots for all feature sets to see the influence of rescoring with different feature sets
- adjusted the readme with new feature sets options

### CODE REVIEW

File: ms2rescore/maxquant.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def _set_mass_error_unit(self) -> None:
         """Get mass error unit from DataFrame columns."""
         if "Mass error [Da]" in self._obj.columns:
             self._mass_error_unit = "Da"
-        elif "Mass Error [ppm]" in self._obj.columns:
+        elif "Mass error [ppm]" in self._obj.columns:
```
Comment (RalfG) [inline ms2rescore/maxquant.py:108]: Problem here is both exist. We need to allow both versions (and potentially also deal with this downstream?)

File: ms2rescore/__init__.py [hunk 2]
```diff
@@ -245,4 +249,34 @@ def run(self):
         if self.config["general"]["run_percolator"]:
             self._run_percolator()
 
+        logger.info("Generating Rescore plots")
+        if self.config["general"]["plotting"]:
+            plotting.PIN(
+                self.config["general"]["output_filename"]
+                + "_"
+                + "_".join(self.config["general"]["feature_sets"][0])
+                + "_features.pin",
```
Comment (RalfG) [inline ms2rescore/__init__.py:258]: Code clarity would improve if this string would be constructed beforehand. Idem below for pout

File: ms2rescore/__init__.py [hunk 3]
```diff
@@ -245,4 +249,34 @@ def run(self):
...
+                    + "_features.pout",
+                    self.config["general"]["output_filename"]
+                    + "_"
+                    + "_".join(fset)
+                    + "_features.pout_dec",
+                    " ".join(fset),
+                    self.config["general"]["output_filename"],
+                )
+
+            plotting._REREC.save_plots_to_pfd(
```
Comment (RalfG) [inline ms2rescore/__init__.py:277]: to pdf instead of `pfd`?

**Review by RalfG — CHANGES_REQUESTED** [review]

**Review by RalfG — CHANGES_REQUESTED** [review]
To do:
- [x] Update `configuration.md`
- [x] Get score name from upstream code? --> Yes: Use PEPREC file instead of PIN, always has `psm_score` column (for retention time).

**Review by paretje — CHANGES_REQUESTED** [review]
I wouldn't have used inheritance of an abstract class, but rather composition, but that can still easily be changed later on if necessary.

**Review by RalfG — APPROVED** [review]

**Pull request merged**